File: app/src/main/java/com/movieapp/huxymovies/adapter/GenreAdapter.java
package com.movieapp.huxymovies.adapter;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.movieapp.huxymovies.R;
import com.movieapp.huxymovies.model.Genre;

import java.util.List;

public class GenreAdapter extends RecyclerView.Adapter<GenreAdapter.GenreHolder> {

    private List<Genre> genreList;

    public GenreAdapter(List<Genre> genres) {
        this.genreList = genres;
    }

    @NonNull
    @Override
    public GenreHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        View layoutView = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.item_genre,
                viewGroup, false);
        return new GenreHolder(layoutView);
    }

    @Override
    public void onBindViewHolder(@NonNull GenreHolder genreHolder, int i) {

        final Genre dm = genreList.get(i);
        genreHolder.genre_txt.setText(dm.getName());
    }

    @Override
    public int getItemCount() {
        return genreList.size();
    }

    static final class GenreHolder extends RecyclerView.ViewHolder

    {

        private TextView genre_txt;

        private GenreHolder(View itemView) {
            super(itemView);
            genre_txt = itemView.findViewById(R.id.genre_txt);
        }
    }
}

File: app/src/main/java/com/movieapp/huxymovies/adapter/GenreAdapter.kt
package com.movieapp.huxymovies.adapter

import android.databinding.DataBindingUtil
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup
import com.movieapp.huxymovies.R
import com.movieapp.huxymovies.databinding.ItemGenreBinding
import com.movieapp.huxymovies.model.Genre

class GenreAdapter(private val genreList: List<Genre>) : RecyclerView.Adapter<GenreAdapter.GenreHolder>() {


    private lateinit var mBinding: ItemGenreBinding
    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): GenreHolder {

        mBinding = DataBindingUtil.inflate(LayoutInflater.from(viewGroup.context),
                R.layout.item_genre, viewGroup, false)

        return GenreHolder(mBinding)
    }

    override fun onBindViewHolder(genreHolder: GenreHolder, i: Int) {

        val dm = genreList[i]
        genreHolder.itemGenreBinding.genre = dm
    }

    override fun getItemCount(): Int {
        return genreList.size
    }

    class GenreHolder constructor(itemView: ItemGenreBinding) : RecyclerView.ViewHolder(itemView.root) {

        var itemGenreBinding: ItemGenreBinding = itemView

    }
}
